core/queryalgebra/model/src/main/java/org/eclipse/rdf4j/query/algebra/AbstractQueryModelNode.java

```diff
@@ -43,6 +43,10 @@ public QueryModelNode getParentNode() {
 
 	@Override
 	public void setParentNode(QueryModelNode parent) {
+		if (this.parent != null && this.parent != parent) {
+			// debug here
+			System.out.println();
```

Comment:
@jeenbroekstra this is a good point to debug when running the tests in QueryPlanRetrievalTest